spiped/dispatch.c

```diff
@@ -136,20 +169,37 @@ callback_gotconn(void * cookie, int s)
 	if ((sas = sock_addr_duplist(A->sas)) == NULL)
 		goto err1;
 
+	/* Create new conn_list_node. */
+	if ((node_new = malloc(sizeof(struct conn_list_node))) == NULL)
+		goto err2;
+	node_new->prev = NULL;
+	node_new->next = NULL;
+
 	/* Create a new connection. */
-	if (proto_conn_create(s, sas, A->decr, A->nofps, A->requirefps,
-	    A->nokeepalive, A->K, A->timeo, callback_conndied, A)) {
+	if ((node_new->conn_cookie = proto_conn_create(s, sas, A->decr,
```

Comment:
If you put a pointer to A inside struct conn_list_node and then passed node_new as the cookie to be passed back to callback_conndied, you wouldn't need to search the linked list.